server/cli/kv.go

```diff
@@ -40,7 +40,8 @@ var osStderr = os.Stderr
 func makeDBClient() *client.DB {
 	// TODO(pmattis): Initialize the user to something more
 	// reasonable. Perhaps Context.Addr should be considered a URL.
-	db, err := client.Open("https://root@" + util.EnsureHost(Context.Addr) +
+	db, err := client.Open(Context.RequestScheme() +
+		"://root@" + util.EnsureHost(Context.Addr) +
 		"?certs=" + Context.Certs)
```

Comment:
Added a test of the `--insecure` flag which revealed an oddity in our command line flags. Now the "kv" parent command defines the `--addr`, `--certs` and `--insecure` flags which its sub-commands inherit.

Yes, the `"?certs=X"` portion of the URL will be ignored if insecure is true.